File: 1452.py
import sys
if(False):file=sys.stdin
else:file=open("input.txt","r")
flag=True
for line in file:
	if(flag):flag=False;continue
	s=list(map(int,line.split()))
file.close()
deltas=[]
for i in range(len(s)):
	for g in range(i+1,len(s)):
		kol=abs(s[i]-s[g])
		if((not kol in deltas) and kol):deltas.append(kol)
output=[1]
ma_kol=1
new_output=[]
for step in deltas:
	new_output=[]
	znach=[]
	kol=1
	for i in range(len(s)-1):
		for g in range(i+1,len(s)):
			if(abs(s[g]-s[i])==step):
				if((s[i] in znach) and (s[g] in znach)):continue
				kol+=1
				if(not s[i] in znach):new_output.append(i+1);znach.append(s[i])
				if(not s[g] in znach):new_output.append(g+1);znach.append(s[g])
				if(ma_kol<kol):output=new_output.copy();ma_kol=kol
print(ma_kol)
print(*output)

File: 1452.c
#include <stdio.h>
#include <stdlib.h>
#include <malloc.h>
#define len 2000
int main(){
	unsigned int i,g,z,h,ma_kol=1,kol,kol_deltas=0,kol_znach,s[len],output[len],znach[len],flag;
	unsigned int* deltas=(unsigned int*)malloc(len*len*sizeof(unsigned int));
	output[0]=1;
	if(1){
		FILE *file=fopen("input.txt","r");
		fscanf(file,"%u\n",&kol);
		for(i=0;i<kol;i++) fscanf(file,"%u ",&s[i]);
		fclose(file);
	}else{
		scanf("%u\n",&kol);
		for(i=0;i<kol;i++) scanf("%u ",&s[i]);
	}
	for(i=0;i<(kol-1);i++){
		for(g=i+1;g<kol;g++){
			if(s[i]!=s[g]){
				deltas[kol_deltas]=s[g]-s[i];
				if(deltas[kol_deltas]<0) deltas[kol_deltas]*=-1;
				flag=1;
				for(z=0;z<kol_deltas;z++){
					if(deltas[z]==deltas[kol_deltas]) {
						flag=0;
						break;
					}
				}
				if(flag) kol_deltas++;
			}
		}
	}
	for(h=0;h<kol_deltas;h++){
		kol_znach=0;
		for(i=0;i<(kol-1);i++){
			for(g=i+1;g<kol;g++){
				if(((s[g]-s[i])==deltas[h]) || ((s[i]-s[g])==deltas[h])){
					flag=0;
					for(z=0;z<kol_znach;z++){
						if(znach[z]==s[i]){
							if(flag==0) flag=1;
							else flag=3;
						}
						if(znach[z]==s[g]){
							if(flag==0) flag=2;
							else flag=3;
						}
					}
					if(flag==3) continue;
					if((flag==2) || (flag==0)){
						znach[kol_znach]=s[i];
						kol_znach++;
					}
					if((flag==1) || (flag==0)){
						znach[kol_znach]=s[g];
						kol_znach++;
					}
					if(ma_kol<kol_znach){
						ma_kol=kol_znach;
						for(z=0;z<kol_znach;z++) output[z]=znach[z];
					}
				}
			}
		}
	}
	printf("%u\n",ma_kol);
	for(i=0;i<ma_kol;i++){
		for(g=0;g<kol;g++) {
			if(s[g]==output[i]){
				printf("%u ",g+1);//Переделать
				break;
			}
		}
	}
	free(deltas);
	//scanf(" ");
	return 0;
}